### PR TITLE
Add support for AutoComplete

### PR DESCRIPTION
**Pull request by Th3-M4jor**
I kept Autocomplete interactions the same inner type as regular `ApplicationCommands` since all it does is add a new field to `ApplicationCommandInteractionDataOption` and an optional field for creating `ApplicationCommand`s.

### CODE REVIEW

File: src/builder/create_interaction_response.rs [hunk 1]
```diff
@@ -148,4 +148,55 @@ impl CreateInteractionResponseData {
         self.0.insert("components", Value::Array(components.0));
         self
     }
+
+    /// For autocomplete responses this sets their autocomlete suggestions.
```
Comment (kangalio) [inline src/builder/create_interaction_response.rs]: Typo "autocomlete"

**Review by kangalio — COMMENTED** [review]
The changes look good to me, but my understanding of the internals of this part of the codebase is limited.

As a more thorough test, I'll play around with autocomplete options a bit once I get home in a few days.

**Comment by kangalio** [comment]
After testing the PR, here's some suggestions:

- an `Interaction::autocomplete()` method, like already exists for the other variants?
- consistently spell it Auto**c**omplete (lowercase c), like Discord itself seems to do
- a designated type for autocomplete interactions? Those methods on ApplicationCommandInteraction don't make sense for autocomplete interactions

**Comment by kangalio** [comment]
Noticed another thing: `CreateInteractionResponseData::add_int_choice` has an i32 parameter even though you're supposed to be able to pass values from -2^53 to 2^53. It should be i64

**Comment by Th3-M4jor** [comment]
Requested changes as discussed in the Discord have been implemented, ready for review.

**Review by arqunis — APPROVED** [review]

**Pull request merged**